### ISSUE TITLE
v2019.07.31-alpha has false positives with schema annotations

### ISSUE THREAD

**Issue by justone** [borkdude/clj-kondo#387]
First off, thank you so much for clj-kondo. It's amazing.

Now for the issue...

**version**

```
$ clj-kondo --version
clj-kondo v2019.07.31-alpha
```

**platform**

Linux, Ubuntu 16.04

My `.clj-kondo/config.edn` file contains:

```
{:lint-as {schema.core/defschema clojure.core/def
           schema.core/defn clojure.core/defn}}
```

**problem**

I upgraded to the latest version of `clj-kondo` and am getting some false positives on some schema annotations.

**repro**

Here is a sample file:

```
(ns sample
  (:require
    [schema.core :as s]
    ))

(s/defschema Person
  {:name s/Str})

(s/defschema Group
  {:people [Person]})

(s/defn group-up :- Group
  [people :- [Person]]
  {:people people})

(s/defn make-people :- [Person]
  []
  [{:name "bob"} {:name "sally"}])

(defn do-it-all
  []
  (group-up (make-people)))

#_(do-it-all)
```

**expected behavior**

clj-kondo should report no errors or warnings.

**actual behavior**

Here is the output from clj-kondo:

```
$ clj-kondo --lang clj --cache --lint sample.clj 
sample.clj:13:11: error: unsupported binding form :-
sample.clj:13:15: warning: unused binding Person
sample.clj:16:25: warning: unused binding Person
sample.clj:22:3: error: sample/group-up is called with 1 arg but expects 3
sample.clj:22:13: error: sample/make-people is called with 0 args but expects 1
linting took 20ms, errors: 3, warnings: 2
```

**Comment by borkdude** [comment]
@justone Can you try removing the config for `schema.core/defn`? clj-kondo has built-in support for it

**Comment by justone** [comment]
Oh wow, that did it! Thank you. I agree that adding defschema to the built-in config would be good.

Thanks again.
